### PR TITLE
fix: remove parameters when resource is deleted or unlinked

### PR DESCRIPTION
**Pull request by jhockett**
#### Description of changes
Add a step a the end of `amplify push` to remove parameters from Parameter Store for any resources that were removed during push.

#### Issue #, if available
Fixes https://github.com/aws-amplify/amplify-cli/issues/12385

#### Description of how you validated changes
e2e test passes locally, also manually verified correct parameters are removed

Passing e2e [run](https://app.circleci.com/pipelines/github/aws-amplify/amplify-cli/19507/workflows/3ae41231-7cfd-45df-9031-cc7d033bc047)

#### Checklist

- [x] PR description included
- [x] `yarn test` passes
- [x] Tests are [changed or added](https://github.com/aws-amplify/amplify-cli/blob/dev/CONTRIBUTING.md#tests)
- [x] [Pull request labels](https://github.com/aws-amplify/amplify-cli/blob/dev/CONTRIBUTING.md#labels) are added

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: packages/amplify-provider-awscloudformation/src/utils/ssm-utils/delete-ssm-parameters.ts [hunk 1]
```diff
@@ -18,16 +18,49 @@ export const deleteEnvironmentParametersFromService = async (context: $TSContext
...
+      }
     }
-    const chunkedKeys: Array<Array<string>> = chunkForParameterStore(envKeysInParameterStore);
+    return false;
+  });
+  await deleteParametersFromParameterStore(client, removedParameterKeys);
+};
+
+const deleteParametersFromParameterStore = async (ssmClient: SSMType, parameterKeys: string[]): Promise<void> => {
+  if (!Array.isArray(parameterKeys) || parameterKeys.length === 0) {
```
Comment (akshbhu) [inline packages/amplify-provider-awscloudformation/src/utils/ssm-utils/delete-ssm-parameters.ts:59]: Do we need to check if its not an Array , if its typed as string[] ?

Comment (jhockett) [reply packages/amplify-provider-awscloudformation/src/utils/ssm-utils/delete-ssm-parameters.ts]: Nope, good catch! Removed.

File: packages/amplify-provider-awscloudformation/src/push-resources.ts [hunk 2]
```diff
@@ -480,7 +480,10 @@ export const run = async (context: $TSContext, resourceDefinition: $TSObject, re
       .map(({ category, resourceName }) => context.amplify.removeDeploymentSecrets(context, category, resourceName));
 
     await adminModelgen(context, resources);
-
+    await deleteEnvironmentParametersForResources(context, context.amplify.getEnvInfo().envName, [
```
[the remark this answers is not in the document]
Comment (jhockett) [reply packages/amplify-provider-awscloudformation/src/push-resources.ts]: That would limit the use of the function to the current environment. I updated to use stateManager to resolve the current envName.

**Review by goldbez — APPROVED** [review]

**Review by akshbhu — APPROVED** [review]

**Review by awsluja — APPROVED** [review]
lgtm

**Pull request merged**